Src/ST-LIB_LOW/Sensors/EncoderSensor/EncoderSensor.cpp

```diff
@@ -16,39 +16,44 @@ void EncoderSensor::start(){
 	uint64_t clock_time = Time::get_global_tick();
 	for(int i = 0; i < N_FRAMES; i++){
 		positions[i] = 0.0;
-		times[i] = i*FRAME_SIZE_IN_SECONDS - N_FRAMES*FRAME_SIZE_IN_SECONDS + (((int) clock_time)/ NANO_SECOND);
+
+		times[i] = i * FRAME_SIZE_IN_SECONDS - N_FRAMES*FRAME_SIZE_IN_SECONDS + (clock_time / NANO_SECOND);
+
 		speeds[i] = 0.0;
 	}
-	time = 0.0;
+	time = clock_time / NANO_SECOND;
 	last_clock_time = clock_time;
 }
 
 void EncoderSensor::read(){
 	optional<uint32_t> optional_counter = Encoder::get_counter(id);
-	optional<bool> direction = Encoder::get_direction(id);
+	optional<bool> optional_direction = Encoder::get_direction(id);
 	uint64_t clock_time = Time::get_global_tick();
 	
-		if(not optional_counter) {
-			ErrorHandler("Error at reading Encoder with id %d counter value", id);
-			return;
-		}
-		else if(not direction) {
-			ErrorHandler("Error at reading Encoder with id %d direction value", id);
-			return;
-		}
-
-	long int delta_clock = clock_time - last_clock_time;
+	if(not optional_counter) {
+		ErrorHandler("Error at reading Encoder with id %d counter value", id);
+		return;
+	}
+	else if(not optional_direction) {
+		ErrorHandler("Error at reading Encoder with id %d direction value", id);
+		return;
+	}
+
+	*direction = (double)optional_direction.value();
+
+	int64_t delta_clock = clock_time - last_clock_time;
 	if(clock_time < last_clock_time){ //overflow handle
 		delta_clock = clock_time + CLOCK_MAX_VALUE * NANO_SECOND / HAL_RCC_GetPCLK1Freq()*2 - last_clock_time;
 	}
```

Comment:
```suggestion
	if(clock_time < last_clock_time) {
		handle_overflow();
	}
```
All my homies hate comments. Ahora en serio, asi te ahorras el comentario, y maskeas la operacion, por si alguien quiere revisar la logica y no le interesa el calculo

Reply:
haces una funcion privada